common/src/main/java/com/microsoft/identity/common/internal/encryption/BrokerEncryptionManager.java

```diff
@@ -0,0 +1,187 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.crypto.SecretKey;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME;
+
+/**
+ * Encryption manager for broker flows.
+ */
+public class BrokerEncryptionManager extends BaseEncryptionManager {
+    private static final String TAG = BrokerEncryptionManager.class.getName();
+
+    /**
+     * A flag to turn on/off keystore encryption on Broker apps.
+     */
+    public static final boolean sShouldEncryptWithKeyStoreKey = false;
+
+    /**
+     * Name of the file contains the keystore-encrypted key for broker.
+     */
+    private static final String BROKERKS = "brokerks";
+
+    /**
+     * Name of the calling package. To be set by test cases only.
+     */
+    private static String sMockPackageName;
+
+    /**
+     * For test cases only. For testing keystore-encrypted key when it's turned off in production.
+     */
+    private boolean mforceEnableKeyStoreKey = false;
+
+    /**
+     * To test legacy keys in common's test project,
+     * we have to set mockPackageName to be one of those broker apps.
+     */
+    public static void setMockPackageName(@NonNull String mockPackageName) {
+        synchronized (BrokerEncryptionManager.class) {
+            sMockPackageName = mockPackageName;
+        }
+    }
+
+    private String getPackageName() {
+        if (sMockPackageName != null) {
+            return sMockPackageName;
+        }
+
+        return mContext.getPackageName();
+    }
+
+    public BrokerEncryptionManager(@NonNull final Context context) {
+        super(context, BROKERKS);
+    }
+
+    // For test cases only.
+    BrokerEncryptionManager(@NonNull final Context context, final boolean forceEnableKeyStoreKey) {
+        super(context, BROKERKS);
+        mforceEnableKeyStoreKey = forceEnableKeyStoreKey;
+    }
+
+    /**
+     * Loads key for Broker.
+     * If new key is enabled and exists, use new key.
+     * Otherwise, use legacy key.
+     */
+    @NonNull
+    @Override
+    public synchronized EncryptionKey loadKeyForEncryption() throws IOException,
+            GeneralSecurityException {
+        final String methodName = ":loadKeyForEncryption";
+
+        if (sShouldEncryptWithKeyStoreKey || mforceEnableKeyStoreKey) {
+            // Try loading existing keystore-encrypted key.
+            // If it doesn't exist nor it is not loadable, fall back to legacy keys.
+            try {
+                final SecretKey key = loadSecretKey(KeyType.KEYSTORE_ENCRYPTED_KEY);
+                if (key != null) {
+                    return new EncryptionKey(KeyType.KEYSTORE_ENCRYPTED_KEY, key);
+                }
+            } catch (final IOException | GeneralSecurityException e) {
+                // If we fail to load key, proceed and generate a new one.
```

Comment:
We are falling back to legacy key, instead of generating a new one. Comment should reflect the same